### PR TITLE
add roles to authorization

### PR DESCRIPTION
**Pull request by annashamray**
Issue - https://github.com/GemeenteUtrecht/ZGW/issues/773 

**Changes:**
* add `Role` to authorizations, which contains the set of permissions
* replace single permission with the role in `BlueprintPermission` 
* since one blueprint is linked now to the set of permissions, remove tight coupling of blueprints and permissions and replace it with tight coupling of bluprints and object types

**Todo (in the next PRs)**:
* expose API endpoints for roles
* add endpoints to create and update authorization profiles

---
**NOTE**

This PR removes all existing blueprints from the DB. 
To recreate the permissions the following steps can be taken: 
* run Django migrations for this PR (this will remove all existing blueprint permissions)
* create roles in the admin 
* execute `python src/manage.py add_blueprint_permissions_for_zaaktypen`
* add created permissions to authorization profiles
---

### CODE REVIEW

**Review by damm89 — APPROVED** [review]
:+1: 
Disclaimer: I skipped through some tests. :no_mouth:

File: backend/doc/authorization.rst [hunk 1]
```diff
@@ -17,6 +17,9 @@ Each permission has the object type the permission relates to. For now three obj
 * document
 * report
 
+Each object type has its required shape for a blueprint permission.
+It is explained in the details in the "Blueprint permission" subsection.
```
Comment (sergei-maertens) [inline backend/doc/authorization.rst:21]: You can label that title in Sphinx and create a clickable link to it.

E.g. use this label setup:

```rst
.. _authorization_blueprints:

Blueprint permissions
---------------------
```

then you can do:

```suggestion
It is explained in the details in :ref:`authorization_blueprints`.
```

File: backend/doc/authorization.rst [hunk 2]
```diff
@@ -87,17 +91,37 @@ The new blueprints can be easily defined for all kinds of objects and their prop
 
       python src/manage.py add_blueprint_permissions_for_zaaktypen
 
+Roles
+^^^^^
+
+Blueprint permissions link the shape of the object (blueprint) with the set of operations.
+Sets of operations represent roles. It is possible to include several (or all) operations in one role.
```
Comment (sergei-maertens) [inline backend/doc/authorization.rst:98]:
```suggestion
Roles represent sets of operations. It is possible to include multiple (or all) operations in one role.
```

File: backend/doc/authorization.rst [hunk 3]
```diff
@@ -87,17 +91,37 @@ The new blueprints can be easily defined for all kinds of objects and their prop
...
+Using roles makes it easy to add the same permissions for different blueprints.
```
Comment (sergei-maertens) [inline backend/doc/authorization.rst:99]:
```suggestion
Using roles simplifies re-using the same permissions for different blueprints.
```

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**